### DISCUSSION TITLE
Incluir swiper.js como dependencia para hacer galería de imágenes

### DISCUSSION THREAD

**Discussion by perealberto** [Aparking/AparKing_Backend#97] [Ideas]
Buenas a todos, estoy desarrollando la vista de detalle de los anuncios de garaje. Para ello, vamos a mostrar las imágenes del garaje en esta vista y permitir que el usuario que la alquila pueda ver el estado del mismo. He estado investigando en ionic que hay una etiqueta que se llama <ion-slide> que te monta un slider de imagenes siguiendo una estructura tan simple como:

```html
<ion-slides [options]="{ slidesPerView: 3, loop: true }">
  <ion-slide>Slide 1</ion-slide>
  <ion-slide>Slide 3</ion-slide>
  <ion-slide>Slide 3</ion-slide>
</ion-slides>
```
Problema, esta característica quedó deprecated en Ionic y la eliminaron. Por lo que no se puede utilizar ahora mismo. En su lugar, en la página oficial animan a usar el framework que estaba detrás de esta etiqueta, [Swiper.js](https://ionicframework.com/docs/ja/v6/angular/slides#properties). Por lo que he estado viendo, no modifica apenas las dependencias y creo que no acarrearía problemas en el despliegue.

Utilizando esta librería, el resultado sería muy similar al anterior:

```html
<swiper-container [slidesPerView]="3" [loop]="true">
  <swiper-slide>Slide 1</swiper-slide>
  <swiper-slide>Slide 2</swiper-slide>
  <swiper-slide>Slide 3</swiper-slide>
</swiper-container>
```

Me gustaría saber si os parece bien que se incluya esta nueva dependencia en el proyecto.

@Aparking/g1 @Aparking/g3

**Comment by sergiosant02** [comment]
Buenas @3ralon, como bien señalas esa característica quedó obsoleta hace algún tiempo. Personalmente me parece un buen aporte, dado que requiere de una instalación realmente simple, efectivamente no supone ningún problema ni mucho menos para el despliegue, es una cambio inocuo.
Muy bien investigado!!!

**Comment by virolialo** [comment]
Me parece buena idea para tratar con poner varias imágenes en los garajes de alquiler :) 
Buen punto!

**Comment by marvicmar** [comment]
La investigación que has realizado @3ralon es genial, tras consultar algunas fuentes el cambio que has propuesto para el manejo del varias imágenes teniendo en cuenta que el otro estaba obsoleto, creo que es la mejor opción y se adecua a nuestras necesidades sin aumentar la complejidad del proyecto. Así que adelante con ello!!
